pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1090,12 +1090,31 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
-// GetAllocatableDevices returns information about all the devices known to the manager
 func (m *ManagerImpl) GetAllocatableDevices() ResourceDeviceInstances {
 	m.mutex.Lock()
-	resp := m.allDevices.Clone()
+	allDevs := m.allDevices.Clone()
+	healthyDevs := cloneHealthDeviceData(m.healthyDevices)
 	m.mutex.Unlock()
-	klog.V(4).InfoS("Known devices", "numDevices", len(resp))
+	klog.V(4).InfoS("Known devices", "numDevices", len(allDevs))
+
+	resp := NewResourceDeviceInstances()
+	for resourceName, deviceInstances := range allDevs {
+		healthyIDs, ok := healthyDevs[resourceName]
+		if !ok {
+			// this is unusual, hence we log with V=2
+			klog.V(2).InfoS("Skipping resource because missing health info", "resource", resourceName)
+			continue
+		}
+		resp[resourceName] = make(map[string]pluginapi.Device)
+		for devID, devInstance := range deviceInstances {
+			if !healthyIDs.Has(devID) {
```

Comment:
shell we consider filtering the m.allocatedDevices

Reply:
Could you please elaborate?

Reply:
As the healthy device maybe still in use, then it's recorded in m.allocatedDevices, it's not allocatable, so should we filter those devices for this method

Reply:
Allocatable in this case means "resources available on the node" not the free resources on the node. Unfortunately we picked a not super-clear name and the context need to be extracted from the original KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2043-pod-resource-concrete-assigments

Reply:
https://github.com/kubernetes/enhancements/pull/2761/commits